### DISCUSSION TITLE
Rider v2.013.1 - Backtest Low Performance

### DISCUSSION THREAD

**Discussion by shichao402** [EA31337/EA31337#408] [Q&A] [answered]
Hello,

I sincerely appreciate all the work you've done so far. I recently got acquainted with MT5 and tried out your project. I attempted a 10-year backtest with the default parameters on the latest release EA31337 Rider v2.013.1 build 1691964469, spanning from 2014 to the present. However, I've noticed that after running for 4.5 hours, the backtest only reached until March 3, 2018. There were numerous warnings in the logs about high spread.

Here are a couple of log entries for reference:

```
2024.05.05 13:30:10.867 Core 01 2018.02.13 08:22:13 2018.02.13 08:22:13: WARNING: EURUSD@M1: Spread too high


2024.05.05 15:13:13.870 Core 01 2018.03.30 12:53:57 2018.03.30 12:53:57: WARNING: EURUSD@M1: Spread too high
```

From the log, it seems to have taken around 1.5 hours to backtest approximately 1.5 months of data. This performance, at least in terms of backtesting, appears to be quite low.

For your information, I am running the latest version of MT5 on an AMD 3600 6-core PC with 16GB of RAM. However, the CPU load seems to be very low, under 10%. Metatester64.exe is using around 7GB of physical memory, leaving some to spare.

I'm wondering if there is something wrong with my settings or if I'm not approaching the test correctly? Is there a way or a guideline to investigate this issue?

Looking forward to your suggestions.

Best regards.

**Comment by kenorb** [comment, marked as answer]
Thanks for the report.
I've noticed it as well with Rider version.
It's something specific to that version, it's performance issue most likely due to large amount of active trades being opened and managed.
For v2.013.1 it has been slightly improved comparing to v2.012 (GH-356).
There is going to be v3.000 released soon, I hope it's already fixed there.
Otherwise it's going to be improved in the following versions.

**Reply by shichao402** [reply]
Understood, thank you for the response. Looking forward to the new version.
